toolchain/check/node_block_stack.h

```diff
@@ -90,9 +101,11 @@ class NodeBlockStack {
   llvm::raw_ostream* vlog_stream_;
 
   // The actual stack.
-  // PushEntry and PopEntry control modification in order to centralize
-  // vlogging.
-  llvm::SmallVector<SemIR::NodeBlockId> stack_;
+  llvm::SmallVector<StackEntry> stack_;
+
+  // The size of the stack. Entries after this in `stack_` are kept around so
+  // that we can reuse the allocated buffer for their content.
+  size_t size_ = 0;
```

Comment:
Given style's preference for int, why size_t?

Reply:
Only to preserve the existing type for `size()`. But it looks like nothing really cares about that, so changed.